src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -873,10 +874,32 @@ protected virtual void Dispose(bool disposing)
 
         public void Dispose()
         {
+            StopWorkerProcessAsync();
             _disposing = true;
             Dispose(true);
         }
 
+        public void StopWorkerProcessAsync()
+        {
+            bool capabilityEnabled = !string.IsNullOrEmpty(_workerCapabilities.GetCapabilityState(RpcWorkerConstants.HandlesWorkerTerminateMessage));
+            if (capabilityEnabled)
+            {
+                var gracePeriod = TimeSpan.FromMilliseconds((double)WorkerConstants.ProcessExitTimeoutInMilliseconds);
```

Comment:
just to confirm, this value is set by each worker? or set by the host? Just wondering if we should be getting this grace period from each worker config

Reply:
Grace period is set by Host. It represents the time Host will wait for worker to exit the process before killing it. I am assigning it the same value which we were already using in the code for this purpose.

https://github.com/Azure/azure-functions-host/blob/9a81e05c957e492c8887a8c2cd180d927f594b35/src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs#L219-L220